circuits/plonk/src/polynomials/varbasemul.rs

```diff
@@ -38,49 +36,49 @@ impl<F: FftField + SquareRootField> ConstraintSystem<F>
                 &(&polys.d8.this.o.scale((2 as u64).into()) - &(&tmp * &polys.d8.this.r)).pow(2)
                 -
                 &(&(&(&polys.d8.this.r.pow(2) - &polys.d8.next.r) + &polys.d8.next.l) * &tmp.pow(2))
-            ).scale(alpha[1])
+            ).scale(alpha[2])
             +
             &(
                 // (xP - xS) × (2*yP - (2*xP - λ1^2 + xT) × λ1) - (yS + yP) * (2*xP - λ1^2 + xT)
                 &(&(&polys.d8.this.l - &polys.d8.next.l) * &(&polys.d8.this.o.scale((2 as u64).into()) - &(&tmp * &polys.d8.this.r)))
                 -
                 &(&(&polys.d8.next.o + &polys.d8.this.o) * &tmp)
-            ).scale(alpha[2]))
+            ).scale(alpha[3]))
             *
             &self.mul2l
         )
     }
 
     // scalar multiplication constraint linearization poly contribution computation
-    pub fn vbmul_scalars(evals: &Vec<ProofEvaluations<F>>, alpha: &Vec<F>) -> Vec<F>
+    pub fn vbmul_scalars(evals: &Vec<ProofEvaluations<F>>, alpha: &[F]) -> Vec<F>
     {
         // 2*xP - λ1^2 + xT
         let tmp = evals[0].l.double() - &evals[0].r.square() + &evals[1].r;
 
         vec!
         [
             // verify booleanity of the scalar bit
-            (evals[0].r.square() - &evals[0].r) * &alpha[1]
+            (evals[0].r.square() - &evals[0].r) * &alpha[0]
```

Comment:
Why the change with the alphas? Were we using higher powers of alpha than necessary?

Reply:
the reason is same as for why we check different equations scaling them with different powers of random value

Reply:
For the record I do not think this is necessary given that the different types of equations are applied on disjoint subsets of the rows. But since it barely affects performance it seems fine to me.